lib/active_model/serializer.rb

```diff
@@ -247,13 +272,17 @@ def json_key
     # Return the +attributes+ of +object+ as presented
     # by the serializer.
     def attributes(requested_attrs = nil)
-      self.class._attributes.each_with_object({}) do |name, hash|
-        next unless requested_attrs.nil? || requested_attrs.include?(name)
-        if self.class._fragmented
-          hash[name] = self.class._fragmented.public_send(name)
-        else
-          hash[name] = send(name)
-        end
+      self.class._attribute_mappings.each_with_object({}) do |(key, attribute_mapping), hash|
```

Comment:
This should probably be memoized. There's no reason I can think of that it would change, right?